content/articles/js-signature-pad/index.md

```diff
@@ -0,0 +1,366 @@
+
+This is the full code:
+
+```javascript
+var canvas = document.getElementById("canvas");
+var saveSignBtn = document.querySelector("[data-action=action-save]");
+var undoBtn = document.querySelector("[data-action=action-undo]");
+var clearBtn = document.querySelector("[data-action=action-clear]");
+var createPDFBtn = document.querySelector("[data-action=action-pdf]");
+
+var ourPad = new SignaturePad(canvas, {
+  backgroundColor: "rgb(255, 255, 255)",
+});
+
+canvas.getContext("2d");
+
+function processImage() {
+  //creating a blob and displaying the image in the image element
+  canvas.toBlob(function (blob) {
+    var targetImg = document.querySelector("img"),
+      url = URL.createObjectURL(blob);
+    targetImg.src = url;
+  });
+}
+
+saveSignBtn.addEventListener("click", function (event) {
+  if (ourPad.isEmpty()) {
+    alert("Please sign first.");
+  } else {
+    processImage();
+  }
+});
+
+//clearing the signature drawing
+clearBtn.addEventListener("click", function (event) {
+  if (ourPad.isEmpty()) {
+    alert("Please sign first.");
+  } else {
+    ourPad.clear();
+  }
+});
+
+undoBtn.addEventListener("click", function (event) {
+  var signMark = ourPad.toData();
+  if (ourPad.isEmpty()) {
+    alert("Please sign first.");
+  } else {
+    if (signMark) {
+      signMark.pop(); // deletinging the last marked dot or drawn line
+      ourPad.fromData(signMark);
+    }
+  }
+});
+
+//creating a pdf using the jsPDF library
+createPDFBtn.addEventListener("click", function (event) {
+  window.html2canvas = html2canvas;
+  window.jsPDF = window.jspdf.jsPDF;
+  var doc = new jsPDF("l", "pt", "a4");
+
+  doc.html(document.getElementById("toPrint"), {
+    callback: function (doc) {
+      doc.save();
+    },
+    x: 10,
+    y: 10,
+  });
+});
+```
+
+### What you can do more with the Signature feature
+
+In the new age of biometrics, much more has been achieved. We have facial recognition, fingerprint recognition, etc. On the downside, what if someone replicates your fingerprint? Remember you cant change it. This means that the person will have lifetime access to your bank details. Also, machines(I group electronics in this category) are never 100% efficient. Let’s say it achieves 99% accuracy but one day something slips and the remaining 1% makes your savings 0? What if, one day, your face deforms(God forbid)? This discussion needs a seperate article anyway.
+
+All these hypotheses lead us to one idea- how about additional authentication(let’s not call it 2-Factor or 3-Factor authentication) when some suspicious activity is detected in your account’s biometric login. A signature can be a good option. I say so because you can alter it at some point, in contrast to a fingerprint that cannot be altered, right? What about utilizing the Signature Pad library and build some form of Machine Learning model(with more inclination to computer vision) to match the signatures for authentication? It can materialize if utilized properly.
```

Comment:
There are a few grammatical errors. Please correct them. Also, let's try to keep it semi-formal

Reply:
Semi-formal? Kindly elaborate.

Reply:
Typically, we avoid using "I", instead we prefer using "we", "us", "let's" and so on

Reply:
Removed this. It wasn't that necessary for the article anyway.